### PR TITLE
Post file reads to concurrent worker from ImmutableBuffer.fromAsset

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards fixing https://github.com/flutter/flutter/issues/107487

Rather than opening the file on the UI thread, post a task to the concurrent worker to read the file. The created SkData is thread safe and can be posted back to the UI thread.

### CODE REVIEW

File: lib/ui/painting/immutable_buffer.cc [hunk 1]
```diff
@@ -40,13 +42,14 @@ Dart_Handle ImmutableBuffer::init(Dart_Handle buffer_handle,
...
 
-Dart_Handle ImmutableBuffer::initFromAsset(Dart_Handle buffer_handle,
+Dart_Handle ImmutableBuffer::initFromAsset(Dart_Handle raw_buffer_handle,
                                            Dart_Handle asset_name_handle,
                                            Dart_Handle callback_handle) {
   UIDartState::ThrowIfUIOperationsProhibited();
   if (!Dart_IsClosure(callback_handle)) {
     return tonic::ToDart("Callback must be a function");
   }
+  std::cerr << "initFromAsset";
```
Comment (dnfield) [inline lib/ui/painting/immutable_buffer.cc:52]: Make this a `TRACE_EVENT`

**Review by dnfield — COMMENTED** [review]
It'd be good to have a test for this.

Maybe you can give UIDartState a special task runner object and assert that initFromAsset scheduled on it?

**Comment by jonahwilliams** [comment]
Isn't loading an existing and non-existing asset a test already?

**Comment by jonahwilliams** [comment]
ehh yeah, unit tests are good too. Lets see if we actually need this

**Comment by jonahwilliams** [comment]
Closing this until we have a reason to implement it...

**Pull request closed without merging**